crates/sparrow-compiler/src/functions/aggregation.rs

```diff
@@ -39,7 +39,7 @@ pub(super) fn register(registry: &mut Registry) {
     registry
         .register("sum<N: number>(input: N, window: window = null) -> N ")
         .with_dfg_signature(
-            "sum<N: number>(input: N, window: window = null, duration: i64 = null) -> N",
+            "sum<N: number>(input: N, window: bool = null, duration: i64 = null) -> N",
```

Comment:
Looking into this. This will cause sparrow tests to fail.

Reply:
This type here is a`window` (and not `boolean`) for type-checking the dfg. The dfg checks that the window argument to the signature is in fact a window, and not just a boolean. After if verifies that and instantiates the types, it flattens the window argument. Additionally, the dfg signature in the registry needs to have the duration argument, so it can include the duration within the pattern.
```
        .with_dfg_signature("sum<N: number>(input: N, window: window = null, duration: i64 = null) -> N")
        .with_implementation(Implementation::new_pattern(&format!(
            "(sum ({}) ({}) ({}))",
            "transform (if ?input_is_new ?input_value) (merge_join ?input_op ?window_op)",
            "?window_value",
            "?duration_value"
```

Reply:
This should work after fixing my signature bugs in https://github.com/kaskada-ai/kaskada/pull/587